### PR TITLE
feat(k8s/refs): add convenience funcs for following k8s references

### PR DESCRIPTION
**Pull request by arschles**
These functions will likely help in the [instance](https://github.com/deis/steward-framework/pull/37) and [binding](https://github.com/deis/steward-framework/pull/33) control loops

cc/ @krancour

### CODE REVIEW

File: k8s/refs/broker_getter_func.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/rest"
+)
+
+// BrokerGetterFunc is the function that attempts to retrieve a broker at the given object ref
+type BrokerGetterFunc func(api.ObjectReference) (*data.Broker, error)
+
+// NewK8sBrokerGetterFunc returns a BrokerGetterFunc  backed by a real kubernetes client
+func NewK8sBrokerGetterFunc(restIface rest.Interface) BrokerGetterFunc {
+	return func(ref api.ObjectReference) (*data.Broker, error) {
```
Comment (krancour) [inline k8s/refs/broker_getter_func.go:15]: My impl took namespace and name as args. This is nicer and cleaner. 👍

File: k8s/refs/broker_getter_func.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		ret := new(data.Broker)
+		url := restutil.AbsPath(
+			restutil.APIVersionBase,
+			restutil.APIVersion,
+			false,
+			ref.Namespace,
+			ref.Name,
+		)
```
Comment (krancour) [inline k8s/refs/broker_getter_func.go:26]: This looks off... I think the pluralized resource type needs to appear in the path. Also, I think that in the call to `restutil.AbsPath(...)`, you are passing `ref.Name` to the argument which is expected (per the function signature) to be the aforementioned `pluralName`.

I believe the following change would correct both these problems:

```
url := append(
  restutil.AbsPath(
    restutil.APIVersionBase,
    restutil.APIVersion,
    false,
    ref.Namespace,
    data.BrokerKindPlural,
  ),
  ref.Name,
)
```

Similar issue exists with the other types.

Alternatively, `restutil.AbsPath(...)` could be refactored. I leave it up to you what you think is the best way to solve for this.

Comment (arschles) [reply k8s/refs/broker_getter_func.go:26]: Good call. Fixed in b0a8dbfb105c9a8173f45e16da8e4cd8e6eff098, along with the other 2 types

**Comment by krancour** [comment]
@arschles do you want to include fake alternatives to these? Or we'll just create those on an as-needed basis? Either is ok with me.

**Comment by arschles** [comment]
@krancour I was on the fence yesterday, thought about it overnight, and I think it's necessary to create them in this PR, in the `./k8s/refs` package, so that both the instance and binding loops can share them. I'll do that now. Commit incoming

**Comment by krancour** [comment]
Ok. I _just_ LGTM'ed. I'll let the LGTM stand despite the incoming amendment.

**Pull request merged**